pkg/gpu/stream.go

```diff
@@ -110,11 +111,12 @@ type kernelSpan struct {
 	avgMemoryUsage map[memAllocType]uint64
 }
 
-func newStreamHandler(pid uint32, sysCtx *systemContext) *StreamHandler {
+func newStreamHandler(pid uint32, containerID string, sysCtx *systemContext) *StreamHandler {
 	return &StreamHandler{
 		memAllocEvents: make(map[uint64]gpuebpf.CudaMemEvent),
 		pid:            pid,
 		sysCtx:         sysCtx,
+		containerID:    containerID,
```

Comment:
is this now part of the unique identifier of the stream? if yes, then let's revive the key struct to contain pid and cid together

Reply:
In theory it isn't, PID+StreamID+GPU_UUID should be the unique identifier.

Reply:
so do you plan to reintroduce the key struct in the multi-gpu PR?

Reply:
Yes, that's the idea.